services/gitea/gitea-pull-requests.service.js

```diff
@@ -0,0 +1,122 @@
+import Joi from 'joi'
+import { pathParam, queryParam } from '../index.js'
+import { optionalUrl, nonNegativeInteger } from '../validators.js'
+import { metric } from '../text-formatters.js'
+import { description, httpErrorsFor } from './gitea-helper.js'
+import GiteaBase from './gitea-base.js'
+
+const schema = Joi.object({ 'x-total-count': nonNegativeInteger }).required()
+
+const queryParamSchema = Joi.object({
+  labels: Joi.string(),
+  gitea_url: optionalUrl,
+}).required()
+
+export default class GiteaPullRequests extends GiteaBase {
+  static category = 'issue-tracking'
+
+  static route = {
+    base: 'gitea/pull-requests',
+    pattern:
+      ':variant(all|all-raw|open|open-raw|closed|closed-raw)/:user/:repo+',
+    queryParamSchema,
+  }
+
+  static openApi = {
+    '/gitea/pull-requests/{variant}/{user}/{repo}': {
+      get: {
+        summary: 'Gitea Pull Requests',
+        description,
+        parameters: [
+          pathParam({
+            name: 'variant',
+            example: 'all',
+            schema: { type: 'string', enum: this.getEnum('variant') },
+          }),
+          pathParam({
+            name: 'user',
+            example: 'gitea',
+          }),
+          pathParam({
+            name: 'repo',
+            example: 'tea',
+          }),
+          queryParam({
+            name: 'gitea_url',
+            example: 'https://gitea.com',
+          }),
+          queryParam({
+            name: 'labels',
+            example: 'test,failure::new',
+            description:
+              'If you want to use multiple labels, you can use a comma (<code>,</code>) to separate them, e.g. <code>foo,bar</code>',
+          }),
+        ],
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'pull requests', color: 'informational' }
+
+  static render({ variant, raw, labels, pullsCount }) {
+    const state = variant
+    const isMultiLabel = labels && labels.includes(',')
+    const labelText = labels ? `${isMultiLabel ? `${labels}` : labels} ` : ''
+
+    let labelPrefix = ''
+    let messageSuffix = ''
+    if (raw) {
+      labelPrefix = `${state} `
+    } else {
+      messageSuffix = state
+    }
+    return {
+      label: `${labelPrefix}${labelText}pull requests`,
+      message: `${metric(pullsCount)}${
+        messageSuffix ? ' ' : ''
+      }${messageSuffix}`,
+      color: pullsCount > 0 ? 'yellow' : 'brightgreen',
+    }
+  }
+
+  async handle(
+    { variant, user, repo },
+    { gitea_url: baseUrl = 'https://gitea.com', labels },
+  ) {
+    const { res } = await this._request(
+      this.authHelper.withBearerAuthHeader({
+        url: `${baseUrl}/api/v1/repos/${user}/${repo}/issues`,
+        options: labels
```

Comment:
I think it would be better to write this as something like

```js
const options = {
  searchParams: {
    page: '1',
    limit: '1',
    type: 'issues',
    state: variant.replace('-raw', ''),
  }
}
if (labels) {
  options.searchParams.labels = labels
}
```

instead of using a ternary. If we duplicate the object, it is hard to tell at a glance what the difference is between them. Also if we want to add another property for example, we'd have to do it in two places.